### PR TITLE
Simplify API for accessing controller's `head`

### PR DESCRIPTION
**Pull request by greg7mdp**
Just provide access to head's `block_handle`, and use `block_handle` member functions to access:
- block_num() 
- timestamp()
- id()
- etc...

This is just a removal of non-necessary access functions, there is no semantic change.

See [discussion](https://github.com/AntelopeIO/spring/pull/316#discussion_r1659398414) with @arhag . 

`block_handle::internal()` was made private in [previous PR](https://github.com/AntelopeIO/spring/pull/316#discussion_r1659398414).

### CODE REVIEW

File: libraries/chain/include/eosio/chain/block_handle.hpp [hunk 1]
```diff
@@ -27,7 +27,8 @@ struct block_handle {
    bool is_valid() const { return !_bsp.valueless_by_exception() && std::visit([](const auto& bsp) { return !!bsp; }, _bsp); }
 
    uint32_t                block_num() const { return std::visit([](const auto& bsp) { return bsp->block_num(); }, _bsp); }
-   block_timestamp_type    block_time() const { return std::visit([](const auto& bsp) { return bsp->timestamp(); }, _bsp); };
+   block_timestamp_type    timestamp() const { return std::visit([](const auto& bsp) { return bsp->timestamp(); }, _bsp); };
+   time_point              block_time() const { return std::visit([](const auto& bsp) { return bsp->timestamp(); }, _bsp); };
```
Comment (heifner) [inline libraries/chain/include/eosio/chain/block_handle.hpp:31]: Explicitly creating the `time_point` will allow for RVO.

**Review by heifner — APPROVED** [review]

**Review by heifner — APPROVED** [review]

File: libraries/chain/include/eosio/chain/block_handle.hpp [hunk 2]
```diff
@@ -27,7 +27,8 @@ struct block_handle {
    bool is_valid() const { return !_bsp.valueless_by_exception() && std::visit([](const auto& bsp) { return !!bsp; }, _bsp); }
 
    uint32_t                block_num() const { return std::visit([](const auto& bsp) { return bsp->block_num(); }, _bsp); }
-   block_timestamp_type    block_time() const { return std::visit([](const auto& bsp) { return bsp->timestamp(); }, _bsp); };
+   block_timestamp_type    timestamp() const { return std::visit([](const auto& bsp) { return bsp->timestamp(); }, _bsp); };
+   time_point              block_time() const { return std::visit([](const auto& bsp) { return time_point{bsp->timestamp()}; }, _bsp); };
```
[the remark this answers is not in the document]
Comment (linh2931) [reply libraries/chain/include/eosio/chain/block_handle.hpp:31]: OK, then no changhes.

**Review by linh2931 — APPROVED** [review]

File: libraries/chain/include/eosio/chain/block_handle.hpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (greg7mdp) [reply libraries/chain/include/eosio/chain/block_handle.hpp:31]: I'm not sure `timepoint` is much clearer than `block_time`, so not changing it in this PR, but I'm not strongly against it either.

**Comment by ericpassmore** [comment]
Note:start
group: CLEANCODE
category: INTERNALS
summary: Simplify API for accessing controller's head.
Note: end

**Pull request merged**